src/utils/typedefs-wrapper.cpp

```diff
@@ -44,26 +46,286 @@ namespace detail {
 }
+
+bool strcomp(Il2CppString const* lhs, const std::u16string_view rhs) noexcept {
+    if (!lhs || lhs->length != (int)rhs.size()) return false;
+
+    Il2CppChar const* first = lhs->chars; 
+    auto* second = rhs.data(); 
+    Il2CppChar const* firstEnd = first + lhs->length; 
+    auto* secondEnd = second + (int)rhs.size(); 
+
+    while (first != firstEnd && second != secondEnd)
+    {
+        if (*first != *second) return false;
+        first++; second++;
+    }
+
+    return first == firstEnd && second == secondEnd;
+}
+
+bool strcomp(Il2CppString const* lhs, const std::string_view rhs) noexcept {
+    if (!lhs || lhs->length != (int)rhs.size()) return false;
+
+    Il2CppChar const* first = lhs->chars; 
+    auto* second = rhs.data(); 
+    Il2CppChar const* firstEnd = first + lhs->length; 
+    auto* secondEnd = second + (int)rhs.size(); 
+
+    while (first != firstEnd && second != secondEnd)
+    {
+        if (*first != *second) return false;
+        first++; second++;
+    }
+
+    return first == firstEnd && second == secondEnd;
+}
+
+bool strless(Il2CppString const* lhs, const std::string_view rhs) noexcept {
+    if (!lhs) return true;
+
+    Il2CppChar const* first = lhs->chars; 
+    char const* second = rhs.data(); 
+    Il2CppChar const* firstEnd = lhs->chars + lhs->length; 
+    char const* secondEnd = rhs.data() + (int)rhs.size(); 
+    
+    while (first != firstEnd && second != secondEnd)
+    {
+        if (*first == *second)
+        {
+            first++; second++;
+            continue;
+        }
+        return *first < *second;
+    }
+    // if we got here, and second is not second end, we had a shorter first, so it should be true
+    // if second is the end, we are longer, so it should be false
+    return second != secondEnd;
+}
+
+bool strless(Il2CppString const* lhs, const std::u16string_view rhs) noexcept {
+    if (!lhs) return true;
+
+    Il2CppChar const* first = lhs->chars; 
+    char16_t const* second = rhs.data(); 
+    Il2CppChar const* firstEnd = first + lhs->length; 
+    char16_t const* secondEnd = second + (int)rhs.size(); 
+    
+    while (first != firstEnd && second != secondEnd)
+    {
+        if (*first == *second)
+        {
+            first++; second++;
+            continue;
+        }
+        return *first < *second;
+    }
+    // if we got here, and second is not second end, we had a shorter first, so it should be true
+    // if second is the end, we are longer, so it should be false
+    return second != secondEnd;
+}
+
+StringW strappend(Il2CppString const* lhs, const std::string_view rhs) noexcept {
+    return std::u16string(lhs->chars, lhs->chars + lhs->length).append(std::u16string(rhs.data(), rhs.data() + rhs.size()));
+}
+
+StringW strappend(Il2CppString const* lhs, const std::u16string_view rhs) noexcept {
+    return std::u16string(lhs->chars, lhs->chars + lhs->length).append(rhs);
+}
+
+bool strstart(Il2CppString const* lhs, const std::string_view rhs) noexcept {
```

Comment:
On a related note, it seems like many of these `str...` functions could be templated to accept either an `std::string_view` or an `std::u16string_view`. Perhaps that's worth considering?